tests/system/test_stream_writer.py

```diff
@@ -0,0 +1,189 @@
+        )
+
+        row_data = WriteStreamSystemTestData()
+        proto_schema = bq_store.WriteStream.gen_proto_schema(row_data)
+        row_data.string_col = "test"
+        row_data.int_col = 0
+        row_data.float_col = 0.1
+        proto_rows = types.ProtoRows()
+        proto_rows.serialized_rows.append(row_data.SerializeToString())
+        (req, resp) = pending_stream.append_rows(rows=proto_rows, schema=proto_schema)
+        assert str(resp.error) == ""
+
+        row_data.string_col = "test2"
+        row_data.int_col = 1
+        row_data.float_col = 0.2
+        proto_rows.serialized_rows.append(row_data.SerializeToString())
+        (req, resp) = pending_stream.append_rows(rows=proto_rows)
+        assert str(resp.error) == ""
+
+        pending_stream.finalize()
+        pending_stream.commit()
+
+    # Test user errors
+    def test_append_rows_invalid_row_data(self, temp_table):
+        default_stream = bq_store.WriteStream(table=temp_table, dataset=TEST_DATASET)
+
+        row_data = WriteStreamSystemTestSchemaErrorData()
+        proto_schema = bq_store.WriteStream.gen_proto_schema(row_data)
+        row_data.wrong_col = "test"
+        proto_rows = types.ProtoRows()
+        proto_rows.serialized_rows.append(row_data.SerializeToString())
+        (req, resp) = default_stream.append_rows(rows=proto_rows, schema=proto_schema)
+        assert resp.code == 400
+        assert isinstance(resp, exceptions.InvalidArgument)
+        assert "no matching field" in str(resp)
+
+    def test_expand_schema_during_write(self, temp_table):
+        table = BQ_CLIENT.get_table(f"{TEST_PROJECT}.{TEST_DATASET}.{temp_table}")
+        original_schema = table.schema
+        new_schema = original_schema[:]
+        new_schema.append(bq.SchemaField("another_float_col", "FLOAT"))
+
+        default_stream = bq_store.WriteStream(table=temp_table, dataset=TEST_DATASET)
+
+        row_data = WriteStreamSystemTestData()
+        proto_schema = bq_store.WriteStream.gen_proto_schema(row_data)
+        row_data.string_col = "test"
+        row_data.int_col = 0
+        row_data.float_col = 0.1
+        proto_rows = types.ProtoRows()
+        proto_rows.serialized_rows.append(row_data.SerializeToString())
+        (req, resp) = default_stream.append_rows(rows=proto_rows, schema=proto_schema)
+        assert str(resp.error) == ""
+
+        table.schema = new_schema
+        table = BQ_CLIENT.update_table(table, ["schema"])
+        time.sleep(1)
+
+        row_data = WriteStreamSystemTestExpandedSchema()
+        proto_schema = bq_store.WriteStream.gen_proto_schema(
+            WriteStreamSystemTestExpandedSchema()
+        )
+        row_data.string_col = "test2"
+        row_data.int_col = 1
+        row_data.float_col = 0.2
+        proto_rows = types.ProtoRows()
+        proto_rows.serialized_rows.append(row_data.SerializeToString())
+        (req, resp) = default_stream.append_rows(rows=proto_rows, schema=proto_schema)
```

Comment:
why do you expect error here?  these seems like a valid schema expansion scenario.